spec/tree_spec.rb

```diff
@@ -1,14 +1,240 @@
 require 'rspec'
 require 'tree'
 
-describe 'Tree' do
-  it 'should be a Class' do
-    expect(described_class.is_a? 'Class').to be_true
+describe Tree do
+  let(:tree) { Tree.new }
+
+  it 'should be an AppleTree' do
+    expect(tree.is_a? Tree).to be true
+  end
+
+  it 'should have a height of 0' do
+    expect(tree.height).to be 0  
+  end
+
+  it 'should have an age of 0' do
+    expect(tree.age).to be 0
+  end
+
+  it 'should have an empty set of apples' do
+    expect(tree.apples).to eq []
+  end
+
+  describe '#age!' do
+    
+    context 'after aging once' do
+      before(:each) { tree.age! }
+
+      it 'age is 1' do
+        expect(tree.age).to be 1
+      end
+
+      it 'height is 1' do
+        expect(tree.height).to be 1
+      end
+
+      it 'has 5 apples' do
+        expect(tree.apples.length).to be 5     
+      end
```

Comment:
Rather then testing the length of the apples array, I think a better approach would've been to mock out the add_apples call and simply verify that it is called. Otherwise, changing the functionality of add_apples (e.g. it only adds 4 apples) results in this test failing when the subject of this test isn't the add_apples at all.